sdk/iot/provisioning/tests/cmocka/test_az_iot_provisioning_client_parser.c

```diff
@@ -0,0 +1,271 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
```

Comment:
Is there a line coverage stats that get generated for ctests in sdk-core? Reason I am asking so as to make sure that our parsing logic is properly tested with all branches covered.

Reply:
Code coverage for this PR can be found [here](https://dev.azure.com/azure-sdk/public/_build/results?buildId=361262&view=codecoverage-tab). 

`az_iot_provisioning_client_c.c` line coverage is at 94.9% and branch coverage is at 63.8%

Reply:
I plan to add more UTs for limit checks and preconditions later.